reed-solomon-novelpoly/src/field/inc_encode.rs

```diff
@@ -200,13 +200,11 @@ pub fn encode_sub_plain(bytes: &[u8], n: usize, k: usize) -> Result<Vec<Additive
 	// pad the incoming bytes with trailing 0s
 	// so we get a buffer of size `N` in `GF` symbols
 	let zero_bytes_to_add = n * 2 - bytes_len;
-	let elm_data = Vec::<Additive>::from_iter(bytes
-		.iter()
-		.copied()
-		.chain(std::iter::repeat(0u8).take(zero_bytes_to_add))
-		.tuple_windows()
-		.step_by(2)
-		.map(|(a, b)| Additive(Elt::from_be_bytes([a, b]))));
+	let mut elm_data = Vec::with_capacity(zero_bytes_to_add + (bytes.len() / 2));
```

Comment:
wait, shouldn't it be `(zero_bytes_to_add + bytes_len) / 2` which is simply `n` (also the comment above and the assert below suggest)? seems we're allocating some extra unused space here

Reply:
indeed 👍🏻 I see you fixed it already